server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -492,17 +493,23 @@ static OverLimitStrategy createOverLimitStrategy(boolean trackRealMemoryUsage) {
         if (trackRealMemoryUsage && jvmInfo.useG1GC().equals("true")
         // messing with GC is "dangerous" so we apply an escape hatch. Not intended to be used.
             && Booleans.parseBoolean(System.getProperty("es.real_memory_circuit_breaker.g1_over_limit_strategy.enabled"), true)) {
-            TimeValue lockTimeout = TimeValue.timeValueMillis(
-                Integer.parseInt(System.getProperty("es.real_memory_circuit_breaker.g1_over_limit_strategy.lock_timeout_ms", "500"))
+
+            long lockTimeoutInMillis = Integer.parseInt(
+                System.getProperty("es.real_memory_circuit_breaker.g1_over_limit_strategy.lock_timeout_ms", "500")
             );
+            TimeValue lockTimeout = TimeValue.timeValueMillis(lockTimeoutInMillis);
+            // TODO: consider a separate property for the lock timeout under full GC
+            TimeValue fullGCLockTimeout = TimeValue.timeValueMillis(lockTimeoutInMillis * 2);
```

Comment:
We'll have to run some experiments to determine the multiplier here. 2x is a reason start. Or maybe you've already done this?

Reply:
Nope, I'm running experiments to tune that as well as the interval